nucleus/src/main/java/za/co/absa/subatomic/application/team/TeamService.java

```diff
@@ -145,4 +157,11 @@ public MembershipRequestEntity findMembershipRequestById(String id) {
         return membershipRequestRepository.findByMembershipRequestId(id);
     }
 
+    @Transactional(readOnly = true)
+    public Set<TeamEntity> findByMemberOrOwner(String slackScreenName) {
+        Set<TeamEntity> teamsWithMemberOrOwner = new HashSet<>();
+        teamsWithMemberOrOwner.addAll(teamRepository.findByMembers_SlackDetailsScreenName(slackScreenName));
+        teamsWithMemberOrOwner.addAll(teamRepository.findByOwners_SlackDetailsScreenName(slackScreenName));
```

Comment:
Should I create query in the repository something like:
```java
List<TeamEntity> findByMembers_SlackDetailsScreenNameOrOwners_SlackDetailsScreenName(String slackScreenName);
```
Instead of the two part query?

Reply:
Nah, let's keep it readable and composable.